instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/HandlerMappingResourceNameFilter.java

```diff
@@ -12,33 +12,44 @@
 import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
 import java.io.IOException;
 import java.util.List;
+import javax.servlet.Filter;
 import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
 import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import org.springframework.beans.factory.support.GenericBeanDefinition;
 import org.springframework.core.Ordered;
-import org.springframework.web.filter.OncePerRequestFilter;
 import org.springframework.web.servlet.HandlerExecutionChain;
 import org.springframework.web.servlet.HandlerMapping;
 
-public class HandlerMappingResourceNameFilter extends OncePerRequestFilter implements Ordered {
+public class HandlerMappingResourceNameFilter implements Filter, Ordered {
```

Comment:
can't have helper class extend `OncePerRequestFilter`, because then `OncePerRequestFilter` won't be instrumented itself